arangod/Aql/ExecutionBlockImpl.cpp

```diff
@@ -1251,6 +1259,7 @@ auto ExecutionBlockImpl<Executor>::executeProduceRows(typename Fetcher::DataRang
     -> std::tuple<ExecutorState, typename Executor::Stats, AqlCall, size_t> {
   if constexpr (isNewStyleExecutor<Executor>) {
     if constexpr (is_one_of_v<Executor, UnsortedGatherExecutor, SortingGatherExecutor, ParallelUnsortedGatherExecutor>) {
+      input.resizeIfNecessary(ExecutorState::HASMORE, 0, _dependencies.size());
```

Comment:
mhm ok, this sounds a bit late too me?

You need to resize if you have not injected enough in the upstream response correct?

Then I would prefer to have the resizing in the Upstream region.
This way you are save against any further usage of the input (e.g. Skip / Produce / Shadow)

Reply:
It's just to avoid running into an assertion in the executor that checks the size; it's not strictly necessary at all. I plan to move this and related code into some `init()` function later.